zerver/lib/events.py

```diff
@@ -705,7 +706,7 @@ def name(sub: Dict[str, Any]) -> str:
             return
 
         old_max_message_id = state['raw_recent_private_conversations'][recipient_id]['max_message_id']
-        if old_max_message_id != event['message_id']:  # nocoverage
+        if old_max_message_id != event['message_ids'][0]:  # nocoverage
```

Comment:
Shouldn't this be an `in` check?

Reply:
I think it's fine either way since this is supposed to be a single private message based on 
```
        # The remainder of this block is about maintaining recent_private_conversations
        if 'raw_recent_private_conversations' not in state or event['message_type'] != 'private':
            return
```
and 
```
    if not sample_message.is_stream_message():
        assert len(messages) == 1
```